### DISCUSSION TITLE
Sigma Rule Validator GitHub Action

### DISCUSSION THREAD

**Discussion by mostafa** [SigmaHQ/sigma#4723] [Show and tell]
Hey,

I just created [this action](https://github.com/marketplace/actions/validate-sigma-rules) based on the JSON schema and scripts (in this repo) and thought I can mention it here. This action helps rule authors to quickly add a validation steps to their repository using GitHub Actions. I also created an [example repository](https://github.com/mostafa/validate-sigma-rules-example) to showcase how it works.

I can donate this tiny project to the Sigma HQ, if you like.

@thomaspatzke 
@nasbench 
@phantinuss 
@kelnage

**Comment by nasbench** [comment]
Hey @mostafa

Cool imitative. But for it to make sense, I would suggest that the action uses the json from the specification repo and the validation script from this sigma repo. This way it would reduce maintenance cost and external dependencies when we update the schema or the validation script.

**Comment by mostafa** [comment]
@nasbench 

Thanks! 🙏 Sounds good. I'll see what I can do.

**Comment by nasbench** [comment]
This is now a public project and hosted in https://github.com/SigmaHQ/sigma-rules-validator
